Jenkinsfile

```diff
@@ -1,28 +1,40 @@
+def cases = [
+    [
+        mavenSettings: [mavenImage: "openjdk:8", serviceAccount: "builder", mavenSettingsXmlSecret: 'm2-settings']
+    ], [
+
+        mavenSettings: [mavenImage: "openjdk:8", serviceAccount: "builder"]
+    ]
+]
+
 node {
     checkout scm
     library identifier: "syndesis-pipeline-library@${env.BRANCH_NAME}", retriever: workspaceRetriever("${WORKSPACE}")
     inNamespace(cloud:'openshift', prefix: 'e2e') {
 
-        stage 'Building'
-        slave {
-            withOpenshift {
-                withMaven(mavenImage: "openjdk:8", serviceAccount: "builder") {
-                    inside {
-                        stage('Prepare Environment') {
-                            createEnvironment(
-                                cloud: 'openshift', name: "${KUBERNETES_NAMESPACE}",
-                                environmentSetupScriptUrl: 'https://raw.githubusercontent.com/syndesisio/syndesis-system-tests/master/src/test/resources/setup.sh',
-                                environmentTeardownScriptUrl: 'https://raw.githubusercontent.com/syndesisio/syndesis-system-tests/master/src/test/resources/teardown.sh',
-                                waitForServiceList: ['syndesis-rest', 'syndesis-ui', 'syndesis-keycloak', 'syndesis-verifier'],
-                                waitTimeout: 600000L,
-                                namespaceDestroyEnabled: false,
-                                namespaceCleanupEnabled: false
-                            )
-                        }
+        stage('Building') {
+            slave {
+                withOpenshift {
+                    for (i = 0; i < cases.size; i++) {
```

Comment:
The position of the loop is a bit confusing and I am not exactly sure what it does internally.
Does it run the build twice, one with each settings pair?
Does it merge the two settings together and just runs the build twice with the merged set?

I would rather see the loop right above the `stage('building')`, so that the scope can be a bit clearer.
Also it doesn't make sense to `Prepare Environment` twice so I would move that outside the loop.